### PR TITLE
[RFR] Show worst move

### PR DESCRIPTION
**Pull request by HALLERPierre**
[Trello](https://trello.com/c/LXlc7VSR/2-05-as-fred-given-its-my-turn-i-want-to-know-if-i-have-losing-moves)

To get the worst move for the player, the client just get the best move for the adversary !

# To do
- [x] Add route and action for worst move
- [x] Add button
- [x] Add highlight for worst move

# Result
prevent-win

### CODE REVIEW

File: app/templates/advice.html.twig [hunk 1]
```diff
@@ -1,21 +1,35 @@
 <a id="advise" class="btn" href="javascript:void(0);">Advise me !</a>
+<a id="worst-advise" class="btn btn-cancel" href="javascript:void(0);">Prevent me to make him win !</a>
```
Comment (alexisjanvier) [inline app/templates/advice.html.twig]: Not sure about the wording ... `What move would make my opponent better?`

File: app/src/Manager/AdvisorManager.php [hunk 2]
```diff
@@ -24,22 +27,52 @@ private function post($body): Response
...
+     *
+     * @return array
+     */
+    public function getWorstAdvice(Game $game): array
+    {
+        $body = [
+            'Grid' => $game->getBoard(),
+            'Player' => $this->gameManager->getEnnemyPlayer($game->getCurrentPlayer()),
+        ];
+        $response = $this->post(json_encode($body));
```
Comment (alexisjanvier) [inline app/src/Manager/AdvisorManager.php]: I don't understand. You make the request to the same endpoint as for `getAdvice`?

File: app/src/Manager/AdvisorManager.php [hunk 3]
```diff
@@ -24,22 +27,52 @@ private function post($body): Response
...
+     *
+     * @param  Game $game
+     *
+     * @return array
+     */
+    public function getWorstAdvice(Game $game): array
+    {
+        $body = [
+            'Grid' => $game->getBoard(),
+            'Player' => $this->gameManager->getEnnemyPlayer($game->getCurrentPlayer()),
```
Comment (alexisjanvier) [inline app/src/Manager/AdvisorManager.php]: Can the worst advice really be the opponent's best advice? I don't understand the logic...

File: app/src/Manager/AdvisorManager.php (hunk 2 above)
Comment (HALLERPierre) [reply app/src/Manager/AdvisorManager.php]: Yes, but for the other player. So the worst move of player 1 is the best move of the player 2

File: app/src/Manager/AdvisorManager.php [hunk 4]
```diff
@@ -24,22 +27,52 @@ private function post($body): Response
...
+    }
+
+    /**
+     * Format stdClass to array
+     *
+     * @param  stdClass $body
+     *
+     * @return array
+     */
+    private function getAdviceFromResponseBody(\stdClass $body): array
```
[the remark this answers is not in the document]
Comment (HALLERPierre) [reply app/src/Manager/AdvisorManager.php]: It's parsed json in the response

File: app/src/Manager/GameManager.php [hunk 5]
```diff
@@ -567,4 +567,22 @@ public function cancelSelection(Game $game): void
...
     }
+
+    /**
+     * getEnnemyPlayer
+     *
+     * @param  int $player
+     *
+     * @return int
+     */
+    public function getEnnemyPlayer(int $player): int
```
[the remark this answers is not in the document]
Comment (HALLERPierre) [reply app/src/Manager/GameManager.php]: Yes, it make more sense ! I'll change that

File: app/src/Manager/AdvisorManager.php (hunk 3 above)
Comment (HALLERPierre) [reply app/src/Manager/AdvisorManager.php]: My worst move is the move that make the opponent win.

File: app/src/Manager/AdvisorManager.php (hunk 2 above)
Comment (HALLERPierre) [reply app/src/Manager/AdvisorManager.php]: Seen together, resolved !

File: app/src/Manager/AdvisorManager.php (hunk 3 above)
Comment (HALLERPierre) [reply app/src/Manager/AdvisorManager.php]: Seen together

File: app/src/Manager/AdvisorManager.php [hunk 6]
```diff
@@ -56,10 +56,10 @@ public function getWorstAdvice(Game $game): array
     {
         $body = [
             'Grid' => $game->getBoard(),
-            'Player' => $this->gameManager->getEnnemyPlayer($game->getCurrentPlayer()),
+            'Player' => $game->getCurrentPlayer(),
         ];
-        $response = $this->post(json_encode($body));
-
+        $response = $this->post(json_encode($body), 'worst-move');
+        dump($response);
```
Comment (alexisjanvier) [inline app/src/Manager/AdvisorManager.php]: debug spotted !

**Pull request merged**